akita/src/internal/crud.ts

```diff
@@ -69,16 +69,28 @@ export class CRUD {
     };
   }
 
-  _update<T extends EntityState>(state: T, ids: ID[], newStateOrFn: object | ((e: Readonly<any>) => object)): T {
+  _update<T extends EntityState>(state: T, ids: ID[], newStateOrFn: object | ((e: Readonly<any>) => object), idKey: string): T {
     const updatedEntities = {};
 
+    let isUpdatingIdKey = false;
+    let idToUpdate: ID;
+
     for (let i = 0; i < ids.length; i++) {
       const id = ids[i];
+      idToUpdate = id;
       assertEntityExists(id, state.entities);
 
       const oldEntity = state.entities[id];
       const newState = isFunction(newStateOrFn) ? newStateOrFn(oldEntity) : newStateOrFn;
 
+      if (newState.hasOwnProperty(idKey) && newState[idKey] !== oldEntity[idKey]) {
```

Comment:
I want to merge it. Can you add a test for this case, please? 

I mean the case when the id is present and the same.

Reply:
sure thing. Will do it soon